### PR TITLE
[Debt] Make action order/colour consistent

### LINKED ISSUE #5738: ✨ Change order pattern of all Confirm/Cancel buttons
## ✨ Feature

Design team has had lengthy discussions about the best approach at ordering buttons at the end of forms and modals, and we have landed on a left to right approach where the leftmost is the most likely/friendly/useful action and the rightmost is the least likely action.

https://www.figma.com/file/G7088DJsXxWY1l4kiGeD1x/Style-guide?type=design&node-id=166-3767&mode=design&t=1BRp0x4onnWBuXN8-0

## 🕵️ Details

This applies to the entire platform and dialogs.

## 📸 Screenshot

image

image

### Pair with Designer
- [ ] Go through app, making a decision about every case
- [ ] Figma updated at same time
### General Direction
- [ ] Update entire app, including dialogs
- [ ] The primary action of the task should use the solid button style
- [ ] Subsequent/secondary actions should use the inline button style
- [ ] The primary action should always come first, left aligned
- [ ] Subsequent/secondary actions should follow the primary action, left aligned
### Button and link colour
- [ ] Main task actions should use the secondary color
- [ ] Actions that cancel a task should use the warning color
- [ ] Actions that are destructive should use the error color
### Uncertainty
- [ ] If unsure how to proceed in a particular case, document it in a new ticket

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #5738 

## 👋 Introduction

Updates all action button and links to have a consistent order and colour scheme.

## 🕵️ Details

@Jerryescandon Helped to settle on the rules for these actions and we worked out the following rules for order and colour.

### Order

1.  Most "helpful" action always first
2. Destructive actions (with no confirmation) appear last
3. Any "neutral" actions appear in the middle in order of helpfulness

### Colour

- Helpful non-destructive action: Secondary (teal)
- Destructive action: Error (red)
- Cancel/return: Warning (yellow)

> [!NOTE]
> This also updated the dialog footer to always place items at the start so we don't need to always override that :sweat_smile: 

## 🧪 Testing

1. Confirm orders in chromatic diff
2. Bounce around different areas in the app where actions are and confirm order

### CODE REVIEW

**Comment by petertgiles** [comment]
@Jerryescandon would you be up for reviewing the Chromatic diff on this one?  It's difficult for a dev to review as there is no design and some interpretation involved.

File: apps/web/src/pages/Applications/ApplicationCareerTimelineAddPage/components/AddExperienceForm.tsx
Comment (petertgiles) [inline apps/web/src/pages/Applications/ApplicationCareerTimelineAddPage/components/AddExperienceForm.tsx:1]: Should this link still be right-justified?
image

**Review by petertgiles — APPROVED** [review]
Code looks good!  I browsed through the site and didn't find any you missed.

File: apps/web/src/pages/PoolCandidates/ViewPoolCandidatePage/components/MoreActions/RevertFinalDecisionDialog.tsx
Comment (petertgiles) [inline apps/web/src/pages/PoolCandidates/ViewPoolCandidatePage/components/MoreActions/RevertFinalDecisionDialog.tsx:1]: Should this button be teal?

File: apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/UpdateSearchRequest.tsx
Comment (petertgiles) [inline apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/UpdateSearchRequest.tsx:1]: Should this one be teal?
image

File: packages/ui/src/components/AlertDialog/AlertDialog.tsx
Comment (petertgiles) [inline packages/ui/src/components/AlertDialog/AlertDialog.tsx:1]: Are these backwards and the wrong colour?
image

File: packages/ui/src/components/Dialog/Dialog.tsx
Comment (petertgiles) [inline packages/ui/src/components/Dialog/Dialog.tsx:1]: Is this the wrong colour?
image

**Review by petertgiles — COMMENTED** [review]
I went through the Chromatic diffs myself, anyway...

File: apps/web/src/pages/PoolCandidates/ViewPoolCandidatePage/components/MoreActions/RevertFinalDecisionDialog.tsx
Comment (esizer) [reply apps/web/src/pages/PoolCandidates/ViewPoolCandidatePage/components/MoreActions/RevertFinalDecisionDialog.tsx:1]: okay, I updated the ones you noted in [fix additional action colours](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10301/commits/d9ac45142360b78040239da7de6212a663964e01)

**Comment by esizer** [comment]
> Looks good! - @Jerryescandon

**Pull request merged**